xds/src/main/java/io/grpc/xds/Bootstrapper.java

```diff
@@ -125,9 +120,163 @@ private static Bootstrap readConfig() throws IOException {
   }
 
   @VisibleForTesting
-  static Bootstrap parseConfig(String rawData) throws InvalidProtocolBufferException {
-    Bootstrap.Builder bootstrapBuilder = Bootstrap.newBuilder();
-    JsonFormat.parser().merge(rawData, bootstrapBuilder);
-    return bootstrapBuilder.build();
+  @SuppressWarnings("unchecked")
+  static BootstrapInfo parseConfig(String rawData) throws IOException {
+    Map<String, ?> rawBootstrap = (Map<String, ?>) JsonParser.parse(rawData);
+
+    Map<String, ?> rawServerConfig = (Map<String, ?>) rawBootstrap.get("xds_server");
```

Comment:
`ServiceConfigUtil.getObject("xds_server")` for better error message on ClassCastException (and avoids unchecked casts). Ditto elsewhere, for all other get()s.

Reply:
Those `ServiceConfigUtil` methods are package private in `io.grpc.internal` though :disappointed_relieved:.

Reply:
Just make them public (just the generic methods). Honestly, I am not a fan of that class. I think we should have a JsonUtil that has those methods in it or some such; I was the one that requested that class, but I wanted it to be _generic_, not data-centric. So also feel free to create a JsonUtil class that makes the methods public.

Reply:
Created #6211 to make them public in a `JsonUtil` class.